pkg/cmd/roachtest/tests/online_restore.go

```diff
@@ -242,6 +153,79 @@ func registerOnlineRestore(r registry.Registry) {
 	}
 }
 
+func registerOnlineRestoreCorrectness(r registry.Registry) {
+	sp := restoreSpecs{
+		hardware: makeHardwareSpecs(hardwareSpecs{workloadNode: true}),
+		backup: makeRestoringBackupSpecs(backupSpecs{
+			nonRevisionHistory: true,
+			version:            fixtureFromMasterVersion,
+			workload:           tpccRestore{opts: tpccRestoreOptions{warehouses: 10, workers: 1, waitFraction: 0}}}),
+		timeout:                15 * time.Minute,
+		suites:                 registry.Suites(registry.Nightly),
+		restoreUptoIncremental: 1,
+		namePrefix:             "online/correctness",
+	}
+	sp.initTestName()
+	r.Add(
+		registry.TestSpec{
+			Name:                sp.testName,
+			Owner:               registry.OwnerDisasterRecovery,
+			Cluster:             sp.hardware.makeClusterSpecs(r, sp.backup.cloud),
+			Timeout:             sp.timeout,
+			CompatibleClouds:    sp.backup.CompatibleClouds(),
+			Suites:              sp.suites,
+			SkipPostValidations: registry.PostValidationReplicaDivergence,
+			Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+				defaultSeed := crdbworkload.NewUint64RandomSeed().Seed()
+				defaultFakeTime := uint32(timeutil.Now().Unix())
+				regRestoreSpecs, regWorkload := initCorrectnessRestoreSpecs(
+					t, sp, defaultSeed, defaultFakeTime, "-reg.trace",
+				)
+				onlineRestoreSpecs, onlineWorkload := initCorrectnessRestoreSpecs(
+					t, sp, defaultSeed, defaultFakeTime, "-online.trace",
+				)
+
+				runRestore(ctx, t, c, regRestoreSpecs, false, true, false)
+				details, err := c.RunWithDetails(
+					ctx,
+					t.L(),
+					option.WithNodes([]int{regRestoreSpecs.hardware.getWorkloadNode()}),
+					fmt.Sprintf("cat %s", regWorkload.opts.queryTraceFile),
+				)
+				require.NoError(t, err, "failed to retrieve query trace for regular restore")
+				regQueryTrace := details[0].Stdout
+
+				c.Wipe(ctx)
+				runRestore(ctx, t, c, onlineRestoreSpecs, true, true, false)
+				details, err = c.RunWithDetails(
+					ctx,
+					t.L(),
+					option.WithNodes([]int{onlineRestoreSpecs.hardware.getWorkloadNode()}),
+					fmt.Sprintf("cat %s", onlineWorkload.opts.queryTraceFile),
+				)
+				require.NoError(t, err, "failed to retrieve query trace for online restore")
+				onlineQueryTrace := details[0].Stdout
+
+				// Compare the query traces.
+				var longerOutput, shorterOutput string
+				if len(regQueryTrace) > len(onlineQueryTrace) {
```

Comment:
how do you ensure that in both workload runs, tpcc runs the same number of queries?

Reply:
seems like we should run x number of queries and then call it day. that would also sidestep the large log file problem you remarked on above.

Reply:
The TPCC workload doesn't provide an option to run a set amount of queries*, so what I do is I run both for a set amount of time, and then check that the output prefix of one run is a prefix of another.

Also, if I could run a set number of queries, I don't think it would circumvent the issue about the log file right? The comparison comes after the workload finishes running.

---

* Well you can set the `max-ops`, so we could set the timeout to some arbitrarily higher value and then choose a max-ops that is guaranteed to be hit?

Reply:
> Well you can set the max-ops, so we could set the timeout to some arbitrarily higher value and then choose a max-ops that is guaranteed to be hit?

yeah, or set max-ops to 10k, observe how long it takes to run normally, and set the timeout to 2x what you observe during the happy path.

Reply:
or maybe, set max-ops to whatever you can get with the number of ops for 5 mintues after online restore completes.